mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -13,13 +13,30 @@ include "mlir/Dialect/LLVMIR/LLVMDialect.td"
 include "mlir/Dialect/LLVMIR/LLVMInterfaces.td"
 include "mlir/IR/AttrTypeBase.td"
 include "mlir/IR/CommonAttrConstraints.td"
+include "mlir/IR/OpAsmInterface.td"
 
 // All of the attributes will extend this class.
 class LLVM_Attr<string name, string attrMnemonic,
                 list<Trait> traits = [],
                 string baseCppClass = "::mlir::Attribute">
-    : AttrDef<LLVM_Dialect, name, traits, baseCppClass> {
+    : AttrDef<LLVM_Dialect, name, traits # [OpAsmAttrInterface], baseCppClass> {
   let mnemonic = attrMnemonic;
+
+  // Custom flag for attribute to indicate whether it wants mnemonic alias.
+  int enableMnemonicAlias = 0;
+  // If the attribute already uses extraClassDeclaration, it should create
+  // its own OpAsmAttrInterface methods instead of relying on here as this
+  // part will be overriden.
+  let extraClassDeclaration = !if(!eq(enableMnemonicAlias, 1), [{
+    //===------------------------------------------------------------------===//
+    // OpAsmAttrInterface Methods
+    //===------------------------------------------------------------------===//
+
+    ::mlir::OpAsmAliasResult getAlias(::llvm::raw_ostream &os) const {
+      os << "}] # mnemonic # [{";
+      return ::mlir::OpAsmAliasResult::OverridableAlias;
+    }
```

Comment:
Ideally, we would always prepend this to the `extraClassDeclaration` in this base class, instead of just setting the entire value. AFAIK, this is not easily doable, without handing the subclass' value to the base class, though.

Reply:
> AFAIK, this is not easily doable, without handing the subclass' value to the base class, though.

I Agree.

Another idea is to modify tablegen to recognize `let alias = xxx` and generate extra code snippet like `extraClassOpAsmAttrInterfaceDeclaration`, but I think that is too much change and may need RFC.

Reply:
Does that imply that all aliases are supposed to implement this function in the exact same way? If so, adding this to tablegen would make sense, I think.

Reply:
No I have a downstream example where the type/attribute want to have custom suffix to better distinguish different instances (instead of the generic 1,2,3 numbering). In this way the type/attribute has to define the function on its own and a tablegen default may not make sense.

https://github.com/google/heir/blob/2c27a9adc27c6ea670e2377d67bc4678d8a3853c/lib/Dialect/LWE/IR/NewLWEAttributes.td#L368-L384

Reply:
I see. In this case this is probably fine then. Opinions @gysit ?

Reply:
It may still make sense to have some tablegen for a flag that can automatically generate a getAlias function that uses the mnemonic. Basically what is in the LLVM attribute base class now. Downstream users that do something more clever can still implement their own solution? 

I would assume that other dialects may also would like to have the same functionality?

However, if this is too complex we can also land the current solution. I am a bit worried that this will lead to unexpected behavior when someone overwrites the extraClassDeclaration. I do not see a better solution though (except for proper tablegen support).

Reply:
Formed a PR on tablegen in #131504. Lets see how people would say about this (namely is it worth it)

Reply:
Thanks for giving it a shot!

Reply:
Now this PR depends on #131504. Pushed a version with that PR cherry-picked to show its effect.

Reply:
Now the dependent PR is merged and this PR is updated.